### PR TITLE
add basic examples

### PR DESCRIPTION
**Pull request by jiacai2050**
# Which issue does this PR close?

Part of #3 

# Rationale for this change
 

Add a basic examples.

# What changes are included in this PR?

1. Add basic example
2. Fix fail_fast, it should return when first error arise.

# Are there any user-facing changes?

# How does this change test

```
cargo run --example basic -- examples/basic.toml
```

### CODE REVIEW

File: examples/basic.toml [hunk 1]
```diff
@@ -0,0 +1 @@
+case_dir = "examples/basic-case"
```
Comment (waynexia) [inline examples/basic.toml]: No empty line in the end

File: src/runner.rs [hunk 2]
```diff
@@ -107,11 +107,11 @@ impl<E: Environment> Runner<E> {
                 Ok(false) => {}
                 Err(e) => {
                     if self.config.fail_fast {
-                        errors.push((case_name, e))
-                    } else {
                         println!("Case {} failed with error {:?}", case_name, e);
                         println!("Stopping environment {} due to previous error.", env);
                         break;
+                    } else {
+                        errors.push((case_name, e))
```
Comment (waynexia) [inline src/runner.rs:114]: Nice catch 🫣

**Review by waynexia — APPROVED** [review]
Looks good to me

File: examples/basic.toml (hunk 1 above)
Comment (jiacai2050) [reply examples/basic.toml]: fixed

**Pull request merged**